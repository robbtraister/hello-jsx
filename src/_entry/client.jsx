'use strict'

import React from 'react'
import ReactDOM from 'react-dom'
import * as ReactRouterDOM from 'react-router-dom'

window.React = React
window.ReactDOM = ReactDOM
window.ReactRouterDOM = ReactRouterDOM

window.render = function (App) {
  const element = window.document.getElementById('app')
  if (!element) return

  if (!App) return

  ReactDOM.render(
    App,
    element
  )
}
