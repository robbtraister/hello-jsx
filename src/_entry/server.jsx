'use strict'

import React from 'react'
import ReactDOM from 'react-dom/server'
// import { StaticRouter } from 'react-router-dom'
import { ServerStyleSheet } from 'styled-components'

// import App from '../app'

function Page (props) {
  // const { context = {}, location, page, ...appProps } = props || {}
  return (
    <html>
      <head>
        <title>Hello JSX</title>
        <link rel='stylesheet' type='text/css' href='/resources/main.css' />
        {/* <link rel='stylesheet' type='text/css' href='/dist/app.css' /> */}
        {/* <script type='application/javascript' src='/dist/runtime/client.js' defer='defer' /> */}
        {/* <script type='application/javascript' src='/dist/vendors~client.js' defer='defer' /> */}
        {/* <script type='application/javascript' src='/dist/libs.js' defer='defer' /> */}
        <script type='application/javascript' src='/dist/runtime.js' defer='defer' />
        <script type='application/javascript' src='/dist/engine.js' defer='defer' />
        <script type='application/javascript' src='/dist/templates/article.js' defer='defer' />
        <dynamic-styles />
        <link rel='icon' type='image/png' href='/resources/icon.png' />
      </head>
      <body>
        <div id='app' />
      </body>
    </html>
  )
}

function render (props) {
  const sheet = new ServerStyleSheet()
  try {
    const html = ReactDOM.renderToStaticMarkup(
      sheet.collectStyles(<Page {...props} />)
    )
    return html
      .replace(
        /<dynamic-styles><\/dynamic-styles>/g,
        sheet.getStyleTags()
      )
  } finally {
    sheet.seal()
  }
}

export {
  Page,
  render
}
