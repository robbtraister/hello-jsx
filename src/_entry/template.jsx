'use strict'

import React from 'react'

import Dark from '../components/layouts/dark'
import Header from '../components/features/header'
import Body from '../components/features/body'

window.Fusion = window.Fusion || {}
window.Fusion.App = (props) =>
  <Dark>
    <Header />
    <Body />
  </Dark>
