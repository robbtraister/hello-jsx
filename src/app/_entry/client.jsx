'use strict'

import React, { lazy, Suspense } from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'

import App from '..'
import Loading from '../components/loading'

const withLazy = (Component) => {
  const LazyComponent = lazy(Component)
  return (props) =>
    <Suspense fallback={<Loading />}>
      <LazyComponent {...props} />
    </Suspense>
}

const Accounts = withLazy(() => import(/* webpackChunkName: "accounts" */ '../pages/accounts'))
const Budget = withLazy(() => import(/* webpackChunkName: "budget" */ '../pages/budget'))
const Guidance = withLazy(() => import(/* webpackChunkName: "guidance" */ '../pages/guidance'))
const Home = withLazy(() => import(/* webpackChunkName: "home" */ '../pages/home'))
const Profile = withLazy(() => import(/* webpackChunkName: "score" */ '../pages/profile'))
const Score = withLazy(() => import(/* webpackChunkName: "score" */ '../pages/score'))

const pages = {
  accounts: Accounts,
  budget: Budget,
  guidance: Guidance,
  home: Home,
  profile: Profile,
  score: Score
}

const Page = (props) =>
  <BrowserRouter>
    <App pages={pages} {...props} />
  </BrowserRouter>

function render (props) {
  ReactDOM.hydrate(
    <Page {...props} />,
    window.document.getElementById('app')
  )
}

window.document.addEventListener('DOMContentLoaded', render)
// window.document.body.onload = render

export { Page }
