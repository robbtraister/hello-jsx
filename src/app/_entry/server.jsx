'use strict'

import React from 'react'
import ReactDOM from 'react-dom/server'
import { StaticRouter } from 'react-router-dom'
import { ServerStyleSheet } from 'styled-components'

import App from '..'

function Page (props) {
  const { context = {}, location, page, ...appProps } = props || {}
  return (
    <html>
      <head>
        <title>Hello JSX</title>
        <link rel='stylesheet' type='text/css' href='/resources/main.css' />
        <link rel='stylesheet' type='text/css' href='/dist/client.css' />
        <script type='application/javascript' src='/dist/client.js' defer='defer' />
        <dynamic-styles />
        <link rel='icon' type='image/png' href='/resources/icon.png' />
      </head>
      <body>
        <div id='app'>
          <StaticRouter context={context} location={location} >
            <App {...appProps} />
          </StaticRouter>
        </div>
      </body>
    </html>
  )
}

function render (props) {
  const sheet = new ServerStyleSheet()
  try {
    const html = ReactDOM.renderToStaticMarkup(
      sheet.collectStyles(<Page {...props} />)
    )
    return html
      .replace(
        /<dynamic-styles><\/dynamic-styles>/g,
        sheet.getStyleTags()
      )
  } finally {
    sheet.seal()
  }
}

export {
  Page,
  render
}
