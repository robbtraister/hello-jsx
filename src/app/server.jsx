'use strict'

import React from 'react'
import ReactDOM from 'react-dom/server'
import { StaticRouter } from 'react-router-dom'
import { ServerStyleSheet } from 'styled-components'

import App from '.'

function Page (props) {
  return (
    <html>
      <head>
        <title>Hello JSX</title>
        <link rel='stylesheet' type='text/css' href='/main.css' />
        <dynamic-styles />
        <script type='application/javascript' src='/client.js' defer='defer' />
        {/* <script type='application/javascript' src={`/${props.app}.js`} defer='defer' /> */}
      </head>
      <body>
        <div id='app'>
          <App Router={StaticRouter} {...props} />
        </div>
      </body>
    </html>
  )
}

function render (props) {
  const sheet = new ServerStyleSheet()
  try {
    const html = ReactDOM.renderToStaticMarkup(
      sheet.collectStyles(
        <Page {...props} />
      )
    )
    return html
      .replace(
        /<dynamic-styles><\/dynamic-styles>/g,
        sheet.getStyleTags()
      )
  } finally {
    sheet.seal()
  }
}

export { render }
