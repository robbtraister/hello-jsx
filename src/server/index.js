#!/usr/bin/env node

'use strict'

const path = require('path')

const express = require('express')

const app = express()

app.use(require('compression')())

app.use('/dist', express.static(path.join(__dirname, '..', '..', 'dist'), { fallthrough: false }))
app.use('/resources', express.static(path.join(__dirname, '..', '..', 'resources'), { fallthrough: false }))

if (/^prod/i.test(process.env.NODE_ENV)) {
  app.use(require('./router'))
} else {
  app.use((req, res, next) => {
    Object.keys(require.cache)
      .filter((mod) => !/\/node_modules\//.test(mod))
      .forEach((mod) => { delete require.cache[mod] })

    require('./router')(req, res, next)
  })
}

const port = process.env.PORT || 8080
app.listen(port, (error) => {
  (error)
    ? console.error(error)
    : console.log(`Listening on port: ${port}`)
})
