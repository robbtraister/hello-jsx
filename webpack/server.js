'use strict'

const path = require('path')

module.exports = [
  {
    ...require('./_shared/devtool'),
    ...require('./_shared/resolve'),
    entry: {
      server: './src/_entry/server'
    },
    module: {
      rules: [
        {
          test: /\.(s[ac]|c)ss$/i,
          use: [
            {
              loader: 'css-loader',
              options: {
                exportOnlyLocals: true,
                modules: true
              }
            }
          ]
        },
        {
          test: /\.s[ac]ss$/i,
          use: 'sass-loader'
        },
        {
          test: /\.m?[jt]sx?$/i,
          exclude: /[\\/]node_modules[\\/]/,
          use: [
            {
              loader: 'babel-loader',
              options: {
                babelrc: false,
                ...require('../babel.config')
              }
            }
          ]
        }
      ]
    },
    output: {
      filename: 'dist/[name].js',
      libraryTarget: 'commonjs2',
      path: path.resolve('.')
    },
    target: 'node'
  }
]
