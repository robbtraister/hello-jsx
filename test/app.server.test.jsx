'use strict'

/* global describe, expect, it */

import React from 'react'
import renderer from 'react-test-renderer'

import { Page } from '../src/app/_entry/server'

describe('snapshots', () => {
  it('render home page', () => {
    const tree = renderer
      .create(<Page location='/home' />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('render score page', () => {
    const tree = renderer
      .create(<Page location='/score' />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('render guidance page', () => {
    const tree = renderer
      .create(<Page location='/guidance' />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
