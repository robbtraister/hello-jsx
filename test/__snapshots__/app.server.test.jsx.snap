// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots render guidance page 1`] = `
<html>
  <head>
    <title>
      Hello JSX
    </title>
    <link
      href="/resources/main.css"
      rel="stylesheet"
      type="text/css"
    />
    <link
      href="/dist/client.css"
      rel="stylesheet"
      type="text/css"
    />
    <script
      defer="defer"
      src="/dist/client.js"
      type="application/javascript"
    />
    <dynamic-styles />
    <link
      href="/resources/icon.png"
      rel="icon"
      type="image/png"
    />
  </head>
  <body>
    <div
      id="app"
    >
      <div
        className="container"
      >
        <div
          className="container"
        >
          <a
            aria-current={null}
            className="tab"
            href="/home"
            onClick={[Function]}
          >
            home
          </a>
          <a
            aria-current={null}
            className="tab"
            href="/score"
            onClick={[Function]}
          >
            score
          </a>
          <a
            aria-current="page"
            className="tab active"
            href="/guidance"
            onClick={[Function]}
            style={Object {}}
          >
            guidance
          </a>
          <a
            aria-current={null}
            className="tab"
            href="/accounts"
            onClick={[Function]}
          >
            accounts
          </a>
          <a
            aria-current={null}
            className="tab"
            href="/budget"
            onClick={[Function]}
          >
            budget
          </a>
          <a
            aria-current={null}
            className="tab profile"
            href="/profile"
            onClick={[Function]}
          >
            Profile
          </a>
        </div>
        <div
          className="body"
        >
          <div
            className="main"
          >
            <div
              className="title"
            >
              Guidance
            </div>
            <div>
              Loading...
            </div>
          </div>
        </div>
        <div
          className="container"
        >
          <div
            className="section"
          >
            Verified
          </div>
          <div
            className="section"
          >
            Contact Us
          </div>
          <div
            className="section"
          >
            Support
          </div>
        </div>
      </div>
    </div>
  </body>
</html>
`;

exports[`snapshots render home page 1`] = `
<html>
  <head>
    <title>
      Hello JSX
    </title>
    <link
      href="/resources/main.css"
      rel="stylesheet"
      type="text/css"
    />
    <link
      href="/dist/client.css"
      rel="stylesheet"
      type="text/css"
    />
    <script
      defer="defer"
      src="/dist/client.js"
      type="application/javascript"
    />
    <dynamic-styles />
    <link
      href="/resources/icon.png"
      rel="icon"
      type="image/png"
    />
  </head>
  <body>
    <div
      id="app"
    >
      <div
        className="container"
      >
        <div
          className="container"
        >
          <a
            aria-current="page"
            className="tab active"
            href="/home"
            onClick={[Function]}
            style={Object {}}
          >
            home
          </a>
          <a
            aria-current={null}
            className="tab"
            href="/score"
            onClick={[Function]}
          >
            score
          </a>
          <a
            aria-current={null}
            className="tab"
            href="/guidance"
            onClick={[Function]}
          >
            guidance
          </a>
          <a
            aria-current={null}
            className="tab"
            href="/accounts"
            onClick={[Function]}
          >
            accounts
          </a>
          <a
            aria-current={null}
            className="tab"
            href="/budget"
            onClick={[Function]}
          >
            budget
          </a>
          <a
            aria-current={null}
            className="tab profile"
            href="/profile"
            onClick={[Function]}
          >
            Profile
          </a>
        </div>
        <div
          className="body"
        >
          <div
            className="main"
          >
            <div
              className="title"
            >
              Home
            </div>
            <div>
              Loading...
            </div>
          </div>
        </div>
        <div
          className="container"
        >
          <div
            className="section"
          >
            Verified
          </div>
          <div
            className="section"
          >
            Contact Us
          </div>
          <div
            className="section"
          >
            Support
          </div>
        </div>
      </div>
    </div>
  </body>
</html>
`;

exports[`snapshots render score page 1`] = `
<html>
  <head>
    <title>
      Hello JSX
    </title>
    <link
      href="/resources/main.css"
      rel="stylesheet"
      type="text/css"
    />
    <link
      href="/dist/client.css"
      rel="stylesheet"
      type="text/css"
    />
    <script
      defer="defer"
      src="/dist/client.js"
      type="application/javascript"
    />
    <dynamic-styles />
    <link
      href="/resources/icon.png"
      rel="icon"
      type="image/png"
    />
  </head>
  <body>
    <div
      id="app"
    >
      <div
        className="container"
      >
        <div
          className="container"
        >
          <a
            aria-current={null}
            className="tab"
            href="/home"
            onClick={[Function]}
          >
            home
          </a>
          <a
            aria-current="page"
            className="tab active"
            href="/score"
            onClick={[Function]}
            style={Object {}}
          >
            score
          </a>
          <a
            aria-current={null}
            className="tab"
            href="/guidance"
            onClick={[Function]}
          >
            guidance
          </a>
          <a
            aria-current={null}
            className="tab"
            href="/accounts"
            onClick={[Function]}
          >
            accounts
          </a>
          <a
            aria-current={null}
            className="tab"
            href="/budget"
            onClick={[Function]}
          >
            budget
          </a>
          <a
            aria-current={null}
            className="tab profile"
            href="/profile"
            onClick={[Function]}
          >
            Profile
          </a>
        </div>
        <div
          className="body"
        >
          <div
            className="main"
          >
            <div
              className="title"
            >
              Wellness Score
            </div>
            <div>
              Loading...
            </div>
          </div>
        </div>
        <div
          className="container"
        >
          <div
            className="section"
          >
            Verified
          </div>
          <div
            className="section"
          >
            Contact Us
          </div>
          <div
            className="section"
          >
            Support
          </div>
        </div>
      </div>
    </div>
  </body>
</html>
`;
